packages/rollup-core/src/app/data/consumers/state-commitment-chain-batch-submitter.ts

```diff
@@ -112,6 +99,40 @@ export class StateCommitmentChainBatchSubmitter extends ScheduledTask {
       return undefined
     }
 
+    return txHash
+  }
+
+  /**
+   * Waits for a confirm to indicate that the transaction did not fail.
+   *
+   * @param txHash The tx hash to wait for.
+   * @param stateRootBatch The rollup batch in question.
+   * @returns true if the tx was successful and false otherwise.
+   */
+  private async waitForProofThatTransactionSucceeded(
+    txHash: string,
+    stateRootBatch: StateCommitmentBatchSubmission
+  ): Promise<boolean> {
+    try {
+      const receipt: TransactionReceipt = await this.stateCommitmentChain.provider.waitForTransaction(
+        txHash,
+        1
```

Comment:
Same note as above WRT >1 block reorgs.  Also noting very similar functionality if there's any possibility to easily share.